DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/streaming_data_interfaces/src/streaming_data_interfaces.cpp

```diff
@@ -0,0 +1,117 @@
+#include <iostream>
+#include <sycl/ext/intel/ac_types/ac_int.hpp>
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include <sycl/ext/intel/prototype/pipes_ext.hpp>
+#include <sycl/sycl.hpp>
+
+#define THRESHOLD 200
+
+// Forward declare the kernel and pipe names
+// (this prevents unwanted name mangling in the optimization report)
+class InPixel;
+class OutPixel;
+class Threshold;
+
+// StreamingBeat struct to support Avalon packet sideband signals
+using StreamingBeatT = sycl::ext::intel::experimental::StreamingBeat<
+    ac_int<16, false>, true, false>;
+
+// A kernel that thresholds pixel values in an image over a stream. Uses start
+// of packet and end of packet signals on the streams to determine the beginning
+// and end of the image.
+template <typename InPixelPipe, typename OutPixelPipe>
+struct ThresholdKernel {
+
+  void operator()() const {
+    bool start_of_packet = false;
+    bool end_of_packet = false;
+
+    while (!end_of_packet) {
+
+      // Read in next pixel
+      StreamingBeatT in_beat = InPixelPipe::read();
+      auto pixel = in_beat.data;
+      start_of_packet = in_beat.sop;
+      end_of_packet = in_beat.eop;
+
+      // Threshold
+      if (pixel > THRESHOLD) pixel = THRESHOLD;
+
+      // Write out result
+      StreamingBeatT out_beat(pixel, start_of_packet, end_of_packet);
+      OutPixelPipe::write(out_beat);
+
+    }
+  }
+};
+
+int main() {
+
+  try {
+
+#if FPGA_SIMULATOR
+  auto selector = sycl::ext::intel::fpga_simulator_selector_v;
+#elif FPGA_HARDWARE
+  auto selector = sycl::ext::intel::fpga_selector_v;
+#else // #if FPGA_EMULATOR
+  auto selector = sycl::ext::intel::fpga_emulator_selector_v;
+#endif
+  sycl::queue q(selector);
+  auto device = q.get_device();
+  std::cout << "Running on device: "
+            << device.get_info<sycl::info::device::name>().c_str()
+            << std::endl;
+
+  // Test image dimensions1
+  unsigned int width  = 16;
+  unsigned int height = 16;
+
+  // Pipe properties
+  using PipePropertiesT = decltype(sycl::ext::oneapi::experimental::properties(
+    sycl::ext::intel::experimental::ready_latency<0>,
+    sycl::ext::intel::experimental::bits_per_symbol<16>,
```

Comment:
Using a 16-bit-wide type while the issue with 8-bit-wide types is being fixed. should I set bits_per_symbol to be 16 to match (so I don't need to add the empty signal)? or should I keep it at 8 and add the empty signal

Reply:
you don't have to add the empty signal, since in this case there will never be an empty symbol.

for tidiness you should probably set bits_per_symbol to 16 though